### PR TITLE
geoDB now allows to share a database with an entire group

### PR DESCRIPTION
**Pull request by thomasstorm**
Closes https://github.com/xcube-dev/xcube-geodb/issues/98

Checklist:

* [x] Issue has been created for change
* [x] Added docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by thomasstorm** [comment]
> ## [Codecov](https://app.codecov.io/gh/xcube-dev/xcube-geodb/pull/99?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=xcube-dev) Report
> Attention: Patch coverage is `87.61905%` with `13 lines` in your changes missing coverage. Please review.

Same as with previous PR: This is due to some necessary Windows-specific changes I made to the test suite. The lines are designed to only run on Windows, so Codecov cannot run them. Therefore I suggest to ignore the Codecov comment.

File: xcube_geodb/core/geodb.py [hunk 1]
```diff
@@ -2391,7 +2391,12 @@ def unpublish_collection_from_group(
 
     def publish_database_to_group(self, group: str, database: Optional[str] = None) -> Message:
         """
-        Publishes the database to the given group.
+        Publishes the database to the given group and enables group users to write into the same database.
+        For example, if some user A creates the database `someproject`, users that share a group G with A are
+        allowed to create new collections in that database (like `someproject_insitudata`) if A publishes the database
+        to the group G. Group users shall not, however, be allowed to read or write any existing collection in the
```
Comment (AliceBalfanz) [inline xcube_geodb/core/geodb.py:2397]: Hmm, why are the other users not allowed to read an existing collection in the shared database?

Comment (thomasstorm) [reply xcube_geodb/core/geodb.py:2397]: > Hmm, why are the other users not allowed to read an existing collection in the shared database?

Methods should do one thing only ("do one thing well", https://en.wikipedia.org/wiki/Unix_philosophy). There is another dedicated method for allowing users to read existing collections in the shared database, i.e. `publish_collection_to_group`. `publish_database_to_group` has been missing before.

**Review by AliceBalfanz — APPROVED** [review]
Thank you, happy to test once deployed :)

**Review by TejasMorbagal — APPROVED** [review]
Looks good :+1:

**Pull request merged**